lib/inspec/control_eval_context.rb

```diff
@@ -12,236 +12,169 @@ module Inspec
-        end
+    def profile_name
+      profile_id
+    end
 
-        define_method :title do |arg|
-          profile_context_owner.set_header(:title, arg)
-        end
+    def control(id, opts = {}, &block)
+      opts[:skip_only_if_eval] = @skip_only_if_eval
 
-        def to_s
-          "Control Evaluation Context (#{profile_name})"
-        end
+      register_control(Inspec::Rule.new(id, profile_id, resources_dsl, opts, &block))
+    end
+    alias rule control
 
-        define_method :profile_name do
-          profile_id
-        end
+    # Describe allows users to write rspec-like bare describe
+    # blocks without declaring an inclosing control. Here, we
+    # generate a control for them automatically and then execute
+    # the describe block in the context of that control.
+    #
+    def describe(*args, &block)
+      loc = block_location(block, caller(1..1).first)
+      id = "(generated from #{loc} #{SecureRandom.hex})"
 
-        define_method :control do |*args, &block|
-          id = args[0]
-          opts = args[1] || {}
-          opts[:skip_only_if_eval] = @skip_only_if_eval
-          register_control(rule_class.new(id, profile_id, opts, &block))
-        end
+      res = nil
+      rule = Inspec::Rule.new(id, profile_id, resources_dsl, {}) do
+        res = describe(*args, &block)
+      end
+      register_control(rule, &block)
 
-        #
-        # Describe allows users to write rspec-like bare describe
-        # blocks without declaring an inclosing control. Here, we
-        # generate a control for them automatically and then execute
-        # the describe block in the context of that control.
-        #
-        define_method :describe do |*args, &block|
-          loc = block_location(block, caller(1..1).first)
-          id = "(generated from #{loc} #{SecureRandom.hex})"
-
-          res = nil
-          rule = rule_class.new(id, profile_id, {}) do
-            res = describe(*args, &block)
-          end
-          register_control(rule, &block)
-
-          res
-        end
+      res
+    end
 
-        define_method :add_resource do |name, new_res|
-          resources_dsl.module_exec do
-            define_method name.to_sym do |*args|
-              new_res.new(@backend, name.to_s, *args)
-            end
-          end
+    def add_resource(name, new_res)
+      resources_dsl.module_exec do
+        define_method name.to_sym do |*args|
+          new_res.new(@backend, name.to_s, *args)
         end
+      end
+    end
 
-        define_method :add_resources do |context|
-          self.class.class_eval do
-            include context.to_resources_dsl
-          end
+    def add_resources(context)
+      # # TODO: write real unit tests for this and then make this change:
+      # dsl = context.to_resources_dsl
+      # self.class.include dsl
+      # Inspec::Rule.include dsl
 
-          rule_class.class_eval do
-            include context.to_resources_dsl
-          end
```

Comment:
Changing 
```
rule_class.class_eval do
  include context.to_resources_dsl
end
```
to
```
Inspec::Rule.class_eval do
  include context.to_resources_dsl
end
```

does not appear to be a not a refactoring, but a data change. Previously each control would end up a with a private copy of the DSL; now that is shared class data on Inspec::Rule.

I am in favor of the change, and removing a layer here is a good thing. I don't know what content out there would try to leverage having a private copy of the resources; but I want to at least have the assurance that some of our most important content won't break because we made what appears to be an innocuous change. 

So, that's why I'm asking for cross-product testing - check for impact on content, just to do due diligence.

Reply:
I'm experimenting with changing it to `extend`... but I have doubts we have any tests that fail if I remove this functionality entirely.